integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/client/TopicSender.java

```diff
@@ -0,0 +1,133 @@
+
+    private static final String initialConnectionFactory = "org.wso2.andes.jndi.PropertiesFileInitialContextFactory";
+    private static final String connectionFactoryNamePrefix = "connectionfactory.";
+    private static final String queueNamePrefix = "topic.";
+    private static final String connectionFactoryName = "andesConnectionfactory";
+
+    private TopicConnection topicConnection;
+    private TopicSession topicSession;
+    private TopicPublisher publisher;
+
+    /**
+     * Initialize the topic publisher
+     * @param topicName Topic name to publish
+     * @param acknowledgeMode JMSAcknowledgement mode
+     * @param configurationReader Server configurations
+     * @throws IOException
+     * @throws NamingException
+     * @throws JMSException
+     */
+    public TopicSender(String topicName, JMSAcknowledgeMode acknowledgeMode,
+         ConfigurationReader configurationReader) throws IOException, NamingException, JMSException {
+
+        // map of config key and config value
+        Map<String, String> clientConfigPropertiesMap = configurationReader.getClientConfigProperties();
+        Properties properties = new Properties();
+        properties.put(Context.INITIAL_CONTEXT_FACTORY, initialConnectionFactory);
+        properties.put(connectionFactoryNamePrefix + connectionFactoryName,
+                getTCPConnectionURL(clientConfigPropertiesMap));
+        properties.put(queueNamePrefix + topicName, topicName);
+        InitialContext ctx = new InitialContext(properties);
+
+        TopicConnectionFactory connectionFactory = (TopicConnectionFactory) ctx.lookup(connectionFactoryName);
+        topicConnection = connectionFactory.createTopicConnection();
+        topicConnection.start();
+        topicSession = topicConnection.createTopicSession(false, acknowledgeMode.getType());
+        Topic topic = (Topic) ctx.lookup(topicName);
+        publisher = topicSession.createPublisher(topic);
+
+    }
+
+    /**
+     * Send a text message to the topic
+     * @param messageCount Number of messages to send
+     * @param textPayload   Message content
+     * @throws JMSException
+     */
+    public void sendMessages(int messageCount, String textPayload) throws JMSException {
+        TextMessage message = topicSession.createTextMessage(textPayload);
+
+        for (int i = 0; i < messageCount; i++) {
+            publisher.publish(message);
+        }
+    }
+
+    /**
+     * Close the publisher connections
+     * @throws JMSException
+     */
+    public void closeSender() throws JMSException {
+        publisher.close();
+        topicSession.close();
+        topicConnection.close();
+    }
+
+    /**
+     * Provide connection URL based on defined parameters.
+     *
+     * @param clientConfigPropertiesMap client connection config properties map
+     * @return connection URL
+     */
+    private String getTCPConnectionURL(Map<String, String> clientConfigPropertiesMap) {
```

Comment:
Cant we move this to a base class ? seems duplicated - https://github.com/wso2/product-ei/pull/904/files#diff-8187b99bcd8aa50e5408434aa8e9c75bR158